RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc

```diff
@@ -305,34 +305,34 @@ void PhotonProducer::fillPhotonCollection(edm::Event& evt,
     int subdet = scRef->seed()->hitsAndFractions()[0].first.subdetId();
     subDetGeometry =  theCaloGeom_->getSubdetectorGeometry(DetId::Ecal, subdet);
 
-    if (subdet==EcalBarrel) { 
+    if (subdet==EcalBarrel) {
       preselCutValues = preselCutValuesBarrel_;
       minR9 = minR9Barrel_;
       hits = ecalBarrelHits;
       flags_ = flagsexclEB_;
       severitiesexcl_ = severitiesexclEB_;
-    } else if  (subdet==EcalEndcap)  { 
+    } else if  (subdet==EcalEndcap)  {
       preselCutValues = preselCutValuesEndcap_;
       minR9 = minR9Endcap_;
       hits = ecalEndcapHits;
       flags_ = flagsexclEE_;
       severitiesexcl_ = severitiesexclEE_;
     } else {
-      edm::LogWarning("")<<"PhotonProducer: do not know if it is a barrel or endcap SuperCluster"; 
+      edm::LogWarning("")<<"PhotonProducer: do not know if it is a barrel or endcap SuperCluster";
     }
+    if(!hits) continue;
```

Comment:
@richard-cms: Better style would be `if (hits == 0)`, since hits is not a boolean, but this way is OK, too.